### PR TITLE
BigAnimal - API Key docs

### PR DESCRIPTION
**Pull request by nidhibhammar**
## What Changed?

Added the API Key folder structure as per [UPM-29649](https://enterprisedb.atlassian.net/browse/UPM-29649).

[UPM-29649]: https://enterprisedb.atlassian.net/browse/UPM-29649?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by xugy99 — COMMENTED** [review]
I don't think the `machine_user.mdx` could belong to `/access_key`.

but instead I think https://www.enterprisedb.com/docs/biganimal/latest/administering_cluster/ is a better parent category

File: product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx [hunk 1]
```diff
@@ -91,3 +91,34 @@ Organization owners can assign users organization-level roles to complete certai
...
+### Add machine user
+
+Only the organization's user having owner role can add a machine user.
+
+To add a machine user:
+
+1. Go to **User Management** from the menu next to your organization name in the top right of the portal.
+1. Select the **Add New User**.
+1. Select the **Machine User** as the user type.
+1. Provide the **Username** and **Email ID**.
```
Comment (zhangcz828) [inline product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx:117]: Email is also optional

File: product_docs/docs/biganimal/release/reference/access_key/index.mdx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+---
+title: "Access key"
+---
+
+Access key is a new authentication process for the BigAnimal users to access the BigAnimal API directly without the `OAuth2` authorization flow. The Access key links to one user only. Each Access key link is immutable since its creation. Each Access Key has an expiration time from one to twelve months (365 days), as customized by users.
+
```
Comment (xugy99) [inline product_docs/docs/biganimal/release/reference/access_key/index.mdx:6]: I think this section probably worth a table like the below screenshot @zhangcz828 found from Azure api key doc:


image

to explicitly tell the difference between normal user's Access key and machine user's access key. 

The table could contain columns like

* user type 
* maximum count (the quota)
* how created and managed

File: product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx [hunk 3]
```diff
@@ -91,3 +91,34 @@ Organization owners can assign users organization-level roles to complete certai
 5.  Select **Submit**.
 
 See [Adding a user to a project](/biganimal/release/administering_cluster/projects.mdx) for information on adding users to projects.
+
+## Machine users
+
+A machine user is used to drive non-GUI access to BigAnimal by user customized automation script, BigAnimal CLI commands, Terraform manifest, BigAnimal's first party binaries that runs from your environment.
+
+- Any user with the organization's owner role can create, update and delete a machine user.
```
Comment (amrita-suresh) [inline product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx:99]: should it be `Owner role` or `owner role`? do we usually capitalized user roles in the BA docs or keep them lowercase?

File: product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx [hunk 4]
```diff
@@ -91,3 +91,34 @@ Organization owners can assign users organization-level roles to complete certai
...
+- The organization's owner user can be a normal user authenticated by IDP via OAuth2 flow or can be a machine user.
```
Comment (amrita-suresh) [inline product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx:100]: The organization's Owner can be either a user authenticated via IDP or a machine user.

File: product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx [hunk 5]
```diff
@@ -91,3 +91,34 @@ Organization owners can assign users organization-level roles to complete certai
...
+- An organization owner can manage the Access key for the machine user but not for the normal users.
+- The organization owner and project owner can assign and unassign the roles to any machine user in the same organization.
+- Machine user is treated the same as the normal user by the BigAnimal AuthZ system on permission check.
+- The machine user is associated to one organization only and can't switch to another organization.
+- The machine user can't be invited.
```
Comment (amrita-suresh) [inline product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx:105]: maybe consolidate with the bullet below?

File: product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx [hunk 6]
```diff
@@ -91,3 +91,34 @@ Organization owners can assign users organization-level roles to complete certai
...
+- The only way to authenticate and authorize a machine user is Access key.
+
+### Add machine user
+
+Only the organization's user having owner role can add a machine user.
```
Comment (amrita-suresh) [inline product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx:110]: Only Organization Owners can add a machine user.

File: product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx [hunk 7]
```diff
@@ -91,3 +91,34 @@ Organization owners can assign users organization-level roles to complete certai
...
+
+To add a machine user:
+
+1. Go to **User Management** from the menu next to your organization name in the top right of the portal.
+1. Select the **Add New User**.
+1. Select the **Machine User** as the user type.
```
Comment (amrita-suresh) [inline product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx:116]: remove "the"

File: product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx [hunk 8]
```diff
@@ -91,3 +91,34 @@ Organization owners can assign users organization-level roles to complete certai
...
+1. Provide the **Username** and **Email ID**.
+1. Optionally, select the **Create Access Key** check box.
+1. Provide the **Access Key Name**.
+1. Select the **Expires in (Day/s)** for the Access key.
+1. Select **Add User** to save the settings.
```
Comment (amrita-suresh) [inline product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx:122]: consolidate with the step below

File: product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx [hunk 9]
```diff
@@ -91,3 +91,34 @@ Organization owners can assign users organization-level roles to complete certai
...
+
+Only the organization's user having owner role can add a machine user.
+
+To add a machine user:
+
+1. Go to **User Management** from the menu next to your organization name in the top right of the portal.
+1. Select the **Add New User**.
+1. Select the **Machine User** as the user type.
+1. Provide the **Username** and **Email ID**.
+1. Optionally, select the **Create Access Key** check box.
```
Comment (amrita-suresh) [inline product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx:119]: maybe we should explain why they would select the check box vs leave it unselected?

File: product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx [hunk 10]
```diff
@@ -91,3 +91,34 @@ Organization owners can assign users organization-level roles to complete certai
...
+1. Provide the **Access Key Name**.
+1. Select the **Expires in (Day/s)** for the Access key.
+1. Select **Add User** to save the settings.
+1. It provides the generated **Access Key** for the user. 
+
+Copy and save this Access Key in a secure location. The Access key is available only at the time you create it. If you lose your Access key, you must delete the Access key and create a new one. For more information, see [Access key](../reference/access_key/).
```
Comment (amrita-suresh) [inline product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx]: this page is inconsistent with capitalizing Access Key - we should decide whether to capitalize it everywhere or not

File: product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx (hunk 3 above)
Comment (nidhibhammar) [reply product_docs/docs/biganimal/release/administering_cluster/01_portal_access.mdx:99]: it's owner role, lowercase across the docs

**Review by amrita-suresh — APPROVED** [review]

**Review by ebgitelman — COMMENTED** [review]
Reviewed in branch and committed. See queries in Slack:
I'm having a hard time figuring out if this is missing an article or a hyphen or if it's okay:
5 per DSS with credit card organization
Can you clarify? 

I also found this a little hard to parse. This is one way I edited it.
- Unlimited for committed-spend organizations that include:
  - CSP marketplace private offer
  - Direct purchase
Is it that the committed spend organizations must have these features? If so, this edit is correct. The other approach I took was this, but on re-reading I thought maybe the above is what was meant:
- Unlimited for committed-spend organizations, which includes:
  - CSP marketplace private offer
  - Direct purchase
In this case, it defines committed-spend organizations as being marketplace private offers or direct purchase. If that's what's meant, then this edit is correct. Let me know which of these two are correct.

File: product_docs/docs/biganimal/release/reference/api.mdx [hunk 11]
```diff
@@ -8,217 +8,22 @@ EDB released v3 of BigAnimal API in January 2023. v2 of BigAnimal API goes out o
...
-To call the BigAnimal API, your application must pass the retrieved access token as a bearer token in the Authorization header of your HTTP request. For example:
+To call the BigAnimal API, your application must pass the retrieved access key as a bearer token in the `x-access-key` header of your HTTP request. For example:
 
 ```shell
 curl --request GET \
-  --url "$AUDIENCE/v3/projects" \
-  --header "authorization: Bearer $ACCESS_TOKEN"  \
-  --header "content-type: application/json"
+  --url "https://portal.biganimal.com/api/v3/projects" \
+  --header "X-ACCESS-KEY: $ACCESS_KEY"
```
Comment (zhangcz828) [inline product_docs/docs/biganimal/release/reference/api.mdx:24]: Can we use lower case for `x-access-key` here to align with the one in line 19?

File: product_docs/docs/biganimal/release/using_cluster/terraform_provider.mdx [hunk 12]
```diff
@@ -32,11 +32,28 @@ To use Terraform with BigAnimal, you need:
 # Configure the BigAnimal Provider
 provider "biganimal" {
   ba_bearer_token = "<redacted>"
-  //ba_api_uri = "https://portal.biganimal.com/api/v3" // Optional
+  // ba_access_key: if set, this will be used instead of the ba_bearer_token above.
+  // This can also be set as an environment variable. If it is set both here and
+  // in an environment variable then the access key set in the environment variable
+  // will take priority and be used
+  ba_access_key = "<redacted>"
+  //ba_api_uri   = "https://portal.biganimal.com/api/v3" // Optional
```
[the remark this answers is not in the document]
Comment (wai-wong-edb) [reply product_docs/docs/biganimal/release/using_cluster/terraform_provider.mdx]: we can put it as `//ba_api_uri   = "<redacted>"`

File: product_docs/docs/biganimal/release/using_cluster/terraform_provider.mdx [hunk 13]
```diff
@@ -32,11 +32,26 @@ To use Terraform with BigAnimal, you need:
...
+  // in an environment variable then the access key set in the environment variable
+  // will take priority and be used
+  ba_access_key = "<redacted>"
 }
 # Manage the resources
 ```
 
+## Environment Variables
+
+Credentials can be provided by using the `BA_BEARER_TOKEN` or `BA_ACCESS_KEY` and optionally `BA_API_URI` environment variables.
```
Comment (wai-wong-edb) [inline product_docs/docs/biganimal/release/using_cluster/terraform_provider.mdx:46]: we can remove BA_API_URI here too

**Review by wai-wong-edb — CHANGES_REQUESTED** [review]

**Review by wai-wong-edb — APPROVED** [review]
